code-challenges.test.js

```diff
@@ -49,6 +82,40 @@ var testArray1 = [3, 7, 10, 5, 4, 3, 3]
 var testArray2 = [7, 8, 2, 3, 1, 5, 4]
 // Expected output: [3, 7, 10, 5, 4, 8, 2, 1]
 
+describe("cleanCombine", () => {
+  it("takes in two arrays as arguments and returns one array with no duplicate values", () => {
+    expect(cleanCombine(testArray1, testArray2)).toEqual([3, 7, 10, 5, 4, 8, 2, 1])
+  })
+})
 
+//Test failed successfully
 
 // b) Create the function that makes the test pass.
+
+// //First I want to create a function with two arrays as inputs
+// const cleanCombine = (array1, array2) => {
+//   //I will combine the two arrays using the spread operator when defining a new variable
+//   let combined = [...array1, ...array2]
+//   //Now that my input arrays are combined into a single array I want to filter out repeating values using the .filter higher order function
+//   return combined.filter((value, index) => {
+//      //I want to filter out any values where the value at an index is equal to the first index of the same value using .indexOf method
+//     return combined.indexOf(value) === index
+//    })
+//  }
+
+//Test passes
+
+//Stretch refactor!
+
+//Define a function that takes in an undefined number of arrays using the spread operator
+const cleanCombine = (...arrays) => {
+  let empty = []
+  // In order to concat all of the arrays, the starting array must be empty, so I defined an empty array. Because .concat is an accessor and not a mutator, and I want to apply other methods to this combined array, I will define a new array (that can be used later) that concatenates the empyty array to the unknown amount of input arrays
+  let combined = empty.concat(...arrays)
+  //Use same methods from first attempt to filter out repeating values across the combined input arrays
+  return combined.filter((value,index) => {
+    return combined.indexOf(value) === index
```

Comment:
YES! indexOf returns the first instance of something so you only get unique values!